spinalcordtoolbox/csa_pmj.py

```diff
@@ -0,0 +1,206 @@
+
+    """
+    native_orientation = Image(segmentation).orientation
+    im_seg = Image(segmentation).change_orientation('RPI')
+    im_pmj = Image(pmj).change_orientation('RPI')
+    nx, ny, nz, nt, px, py, pz, pt = im_seg.dim
+    pr = min([px, py])
+    im_segr = resample_nib(im_seg, new_size=[pr, pr, pz], new_size_type='mm', interpolation='linear')
+    im_pmjr = resample_nib(im_pmj, new_size=[pr, pr, pz], new_size_type='mm', interpolation='linear')
+    data_pmj = im_pmjr.data
+    # Update dimensions from resampled image
+    nx, ny, nz, nt, px, py, pz, pt = im_segr.dim
+
+    # Extract min and max index in Z direction
+    data_seg = im_segr.data
+    X, Y, Z = (data_seg > NEAR_ZERO_THRESHOLD).nonzero()
+    min_z_index = min(Z)
+    # Compute the spinal cord centerline based on the spinal cord segmentation
+    im_ctl, arr_ctl, arr_ctl_der, fit_results = get_centerline(im_segr, param=param_centerline, verbose=verbose)
+
+    # Extrapolate centerline
+    arr_ctl_extra = extrapolate_centerline(arr_ctl, im_segr, pz)
+
+    # Create an image with the centerline | ONLY TO VALIDATE --> to remove
+    im_centerline = im_segr.copy()
+    im_centerline.data = np.zeros(im_centerline.data.shape)
+    # Assign value=1 to centerline. Make sure to clip to avoid array overflow.
+    im_centerline.data[round_and_clip(arr_ctl_extra[0], clip=[0, im_centerline.data.shape[0]]),
+                       round_and_clip(arr_ctl_extra[1], clip=[0, im_centerline.data.shape[1]]),
+                       np.array(range(im_segr.dim[2]))] = 1
+    im_centerline.save('centerline_extrapolated.nii.gz')
+    pmj_coord = np.argwhere(data_pmj != 0)[0]
+    # Get Z index of PMJ project on extrapolated centerline
+    pmj_index = get_min_distance(pmj_coord, arr_ctl_extra, px, py, pz)
+    # Compute distance from PMJ along centerline
+    arr_length = get_distance_from_pmj(arr_ctl_extra, pmj_index, px, py, pz)
+
+    # Check if distance is out of bound
+    if distance > arr_length[0][0]:
+        raise ValueError("Input distance of " + str(distance) + " mm is out of bound for maximum distance of " + str(arr_length[0][0]) + " mm")
+
+    if distance < arr_length[0][-1]:
+        raise ValueError("Input distance of " + str(distance) + " mm is out of bound for minimum distance of " + str(arr_length[0][-1]) + " mm")
+
+    # Get Z index of corresponding distance from PMJ with the specified extent
+    z_index_extent_min = get_nearest_index(arr_length, distance + extent/2)
+    z_index_extent_max = get_nearest_index(arr_length, distance - extent/2)
+    # Check if extent corresponds to the lenght, if not add or remove a slice
+    z_index_extent_min, z_index_extent_max = validate_length(z_index_extent_min, z_index_extent_max, arr_length, extent)  # Find a quicker way to solve this
+    # Check if min Z index is available in the segmentation, if not, use the min_z_index of segmentation
+    if z_index_extent_min < min_z_index:
+        z_index_extent_min = min_z_index
```

Comment:
you might want to add a warning to let the user know that the requested I-S length will be considered because...

Reply:
done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3429/commits/387ee6962b903e1d3495b0fbae1c689ae800884a